### PR TITLE
Refactor endpoint service

### PR DESCRIPTION
**Pull request by danvixent**

### CODE REVIEW

File: services/create_endpoint.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	Cache        cache.Cache
+	EndpointRepo datastore.EndpointRepository
+	ProjectRepo  datastore.ProjectRepository
+
+	E         models.Endpoint
+	ProjectID string
+}
+
+func (a *CreateEndpointService) Run(ctx context.Context) (*datastore.Endpoint, error) {
+	if err := util.Validate(a.E); err != nil {
```
Comment (subomi) [inline services/create_endpoint.go:29]: @danvixent As a rule of thumb, model validations should happen in the handler.  Only other nuanced core business logic validations should happen here.

File: services/endpoint_service.go [hunk 2]
```diff
@@ -1,23 +1,9 @@
 package services
```
Comment (subomi) [inline services/endpoint_service.go:1]: @danvixent Do we still need this file?

**Review by subomi — CHANGES_REQUESTED** [review]

File: services/endpoint_service.go (hunk 2 above)
Comment (danvixent) [reply services/endpoint_service.go]: @subomi I removed it

File: services/create_endpoint.go (hunk 1 above)
Comment (danvixent) [reply services/create_endpoint.go]: @subomi fixed

Comment (danvixent) [reply services/create_endpoint.go]: @subomi fixed

File: services/endpoint_service.go (hunk 2 above)
Comment (danvixent) [reply services/endpoint_service.go]: @subomi I removed it

**Review by subomi — APPROVED** [review]

**Pull request merged**